lib/Hakyll/Core/Dependencies.hs

```diff
@@ -88,6 +106,19 @@ markOod id' = State.modify $ \s ->
     s {dependencyOod = S.insert id' $ dependencyOod s}
 
 
+--------------------------------------------------------------------------------
+data DependencyKind = KindContent | KindMetadata
+  deriving (Show)
+
+instance Binary DependencyKind where
+  put KindContent = putWord8 0
+  put KindMetadata = putWord8 1
+
+  get = getWord8 >>= \t -> case t of
+      0 -> pure KindContent
+      1 -> pure KindMetadata
+      _ -> error "Data.Binary.get: Invalid DependencyKind"
+
```

Comment:
How do we ensure that the binary cache is clearly broken by this addition, and does not remain incoherent but still parseable?

One way would be to replace the tokens `0` and `1` by something more explicit, but larger, such as `"kindContent"` and `"kindMetadata"`, but maybe I'm being overly cautious.